onnxruntime/contrib_ops/cuda/bert/attention.cc

```diff
@@ -135,8 +135,31 @@ Status Attention<T>::ComputeInternal(OpKernelContext* context) const {
   if (use_flash_attention && parameters.sequence_length < min_seq_len_for_flash_attention_packed_qkv_) {
     use_flash_attention = false;
   }
+  // Allocate buffers
```

Comment:
This change is quite similar to the change in MHA. Please add a helper function to avoid duplicate the code?